cmd/snap/main.go

```diff
@@ -404,6 +417,19 @@ func run() error {
 		}
 
 		msg, err := errorToCmdMessage("", err, nil)
+
+		if cmdline := strings.Join(os.Args, " "); strings.ContainsAny(cmdline, wrongDashes) {
```

Comment:
How do you feel about doing a `strings.Replace` before we show any of that to the argument parser. If we _know_ what the user wanted can we just do it?

Reply:
I was tempted by it, but it seems like a slippery slope we don't want to slither down

Reply:
to elaborate on this a  bit: we couldn't just use `strings.Replace`, because then a user wouldn't be able to use em dashes in a `snap set`, for example; we'd have to actually parse it.